proteus/WaveTools.py

```diff
@@ -0,0 +1,241 @@
+
+class RandomWaves:
+    """Generate approximate random wave solutions
+
+    :param Tp: peak period [T]
+    :param Hs: significant wave height [L]
+    :param  d: depth [L]
+    :param fp: frequency [1/T]
+    :param bandFactor: width factor for band  around fp [-]
+    :param N: number of frequency bins [-]
+    :param mwl: mean water level [L]"""
+    
+    def __init__(self,
+                 Tp = 5.0,         #s peak period
+                 Hs = 2.0,         #m significant wave height
+                 d = 2.0,           #m depth
+                 fp = 1.0/5.0,      #peak  frequency
+                 bandFactor = 2.0, #controls width of band  around fp
+                 N = 101,          #number of frequency bins
+                 mwl = 0.0,        #mean water level
+                 g = 9.81,         #accelerationof gravity
+                 spec_fun = JONSWAP): #wave spectrum
+        self.Tp = Tp
+        self.Hs = Hs
+        self.d = d
+        self.fp = fp
+        self.bandFactor = bandFactor
+        self.N = N
+        self.mwl = mwl
+        self.g = g
+        self.fmax = self.bandFactor*self.fp
+        self.fmin = self.fp/self.bandFactor
+        self.df = (self.fmax-self.fmin)/float(self.N-1)
+        self.fi=np.zeros(self.N,'d')
+        for i in range(self.N):
+            self.fi[i] = self.fmin+self.df*i
+        self.ki = dispersion(2.0*pi*self.fi,self.d,g=self.g)
+        self.phi = 2.0*pi*np.random.random(self.fi.shape[0])
+        #ai = np.sqrt((Si_J[1:]+Si_J[:-1])*(fi[1:]-fi[:-1]))
+        fim_tmp = (0.5*(self.fi[1:]+self.fi[:-1])).tolist()
+        self.fim = np.array([fim_tmp[0]-0.5*self.df]+fim_tmp+[fim_tmp[-1]+0.5*self.df])
+        self.Si_Jm = spec_fun(self.fim,f0=self.fp,Hs=self.Hs,g=self.g,gamma=3.3)
+        self.ai = np.sqrt((self.Si_Jm[1:]+self.Si_Jm[:-1])*(self.fim[1:]-self.fim[:-1]))
+    
+    def eta(self,x,t):
+        """Free surface displacement
+        
+        :param x: floating point x coordinate
+        :param t: time"""
+        return (self.ai*np.cos(2.0*pi*self.fi*t - self.ki*x + self.phi)).sum()
+    
+    def u(self,x,z,t):
+        """x-component of velocity
+
+        :param x: floating point x coordinate
+        :param z: floating point z coordinate (height above bottom)
+        :param t: time
+        """
+        Z = z - self.mwl
+        return (2.0*pi*self.fi*self.ai*np.cos(2.0*pi*self.fi*t-self.ki*x+self.phi)*
+                np.cosh(self.ki*(self.d+Z))/np.sinh(self.ki*self.d)).sum()
+    
+    def w(self,x,z,t):
+        """z-component of velocity
+        
+        :param x: floating point x coordinate
+        :param z: floating point z coordinate (height above bottom)
+        :param t: time"""
+        Z = z - self.mwl
+        return (2.0*pi*self.fi*self.ai*np.sin(2.0*pi*self.fi*t-self.ki*x+self.phi)*
+                np.sinh(self.ki*(self.d+Z))/np.sinh(self.ki*self.d)).sum()
+
+if __name__ == '__main__':
+    Tp = 5.0 #s peak period
```

Comment:
All of this should go into a function.  It will be easier to run/test/reuse if it doesn't require executing the script as main.